CSharpSourceCode/HarmonyPatches/MissionPatches.cs

```diff
@@ -0,0 +1,71 @@
+﻿using HarmonyLib;
+using System.Xml;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Core;
+using TaleWorlds.ModuleManager;
+using TaleWorlds.MountAndBlade;
+using TaleWorlds.ObjectSystem;
+using TOW_Core.Utilities.Extensions;
+
+[HarmonyPatch]
+public static class MissionPatches
+{
+    [HarmonyPrefix]
+    [HarmonyPatch(typeof(Mission), "SpawnAgent")]
+    public static bool SpawnAgentPrefix(AgentBuildData agentBuildData)
+    {
+        var character = agentBuildData.AgentCharacter as CharacterObject;
+        if (character != null)
+        {
+            string name = null;
+            if (character.HeroObject != null && character.HeroObject.IsVampire())
+            {
+                name = "vampire";
+            }
+            else if (character.IsVampire())
+            {
+                name = "vampire";
+            }
+            if (!string.IsNullOrWhiteSpace(name))
+            {
+                Monster monster = GetCustomMonster(name);
```

Comment:
maybe cache this Monster object in a class level static field to avoid running IO operation (xml read) every time an agent is spawned. Faster load time that way I think.

Reply:
Yeah, you're right

Reply:
Fixed